python/packages/azurefunctions/agent_framework_azurefunctions/_orchestration.py

```diff
@@ -144,7 +164,57 @@ def my_orchestration(context):
 
         # Call the entity and return the Task directly
         # The orchestration will yield this Task
-        return self.context.call_entity(entity_id, "run_agent", run_request.to_dict())
+        result = yield self.context.call_entity(entity_id, "run_agent", run_request.to_dict())
+
+        logger.debug(
+            "[DurableAIAgent] Entity call completed for correlation_id %s",
+            correlation_id,
+        )
+
+        response = self._load_agent_response(result)
+
+        if response_format is not None:
+            self._ensure_response_format(response_format, correlation_id, response)
+
+        return response
+
+    def _load_agent_response(self, agent_response: AgentRunResponse | dict[str, Any] | None) -> AgentRunResponse:
+        """Convert raw payloads into AgentRunResponse instance."""
+        if agent_response is None:
+            raise ValueError("agent_response cannot be None")
+
+        logger.debug(f"[load_agent_response] Loading agent response of type: {type(agent_response)}")
```

Comment:
NIT: Can we not use f strings for logging? There is a perf hit with logging and fstrings. I had to change all the python worker logs from fstring to lazy logging because of this.

Reply:
Maybe we need to do that for the whole package. We can do it in another task

Reply:
Thanks for the catch, I wasnt aware of the perf impact. 
But I agree, we can do this as a separate task.